### ISSUE TITLE
Build error in Docker ubuntu-22.04

### ISSUE THREAD

**Issue by justas05** [aws/credentials-fetcher#125]
```
/root/credentials-fetcher/api/src/gmsa_service.cpp:4:10: fatal error: 'credentialsfetcher.grpc.pb.h' file not found
#include <credentialsfetcher.grpc.pb.h>
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 warning and 1 error generated.
make[2]: *** [api/CMakeFiles/cf_gmsa_service_private.dir/build.make:79: api/CMakeFiles/cf_gmsa_service_private.dir/src/gmsa_service.cpp.o] Error 1
make[2]: Leaving directory '/root/credentials-fetcher/build'
make[1]: *** [CMakeFiles/Makefile2:247: api/CMakeFiles/cf_gmsa_service_private.dir/all] Error 2
make[1]: Leaving directory '/root/credentials-fetcher/build'
make: *** [Makefile:139: all] Error 2
```

**Comment by jamolina** [comment]
Thank you for reaching out @justas05 . We will take a look at the error and get back to you. One question, do you see this error when cloning the repo and without making any changes to the code? Or do you see this only on your PR branch?

**Comment by justas05** [comment]
This is the next error. I commented out lines 186 to 191 in the file
https://github.com/aws/credentials-fetcher/blob/mainline/CMakeLists.txt to
disable tests

**Comment by justas05** [comment]
@jamolina, were you able to watch this?

**Comment by jamolina** [comment]
Hey @justas05. I believe @bhallasaksham just pin pointed the issue. We expect to send a fix soon.

**Comment by jamolina** [comment]
@justas05 could you try again? @bhallasaksham just pushed a fix for this

**Comment by justas05** [comment]
@jamolina, thank you! It seems the problem has been resolved. But the first time I ran it I got an error with `make -j`.
```
In file included from /root/credentials-fetcher/api/src/gmsa_service.cpp:4:
/root/credentials-fetcher/build/api/credentialsfetcher.grpc.pb.h:7:10: fatal error: credentialsfetcher.pb.h: No such file or directory
    7 | #include "credentialsfetcher.pb.h"
```
I then tried running it with make and the build completed successfully. Perhaps there is a race somewhere.

**Comment by jamolina** [comment]
We could not reproduce the error. What are you using for the `TIME_ZONE` argument?

**Comment by justas05** [comment]
This error occurred when building on Ubuntu without Docker. I did not set the TIME_ZONE value.

**Comment by jamolina** [comment]
We tested this in a couple of Ubuntu instances and it worked successfully the first time. Since the original issue has been resolved and your PR is unblocked, we can close this issue for now. If you encounter it again and find a way to reproduce it, feel free to open a new issue.

**Comment by justas05** [comment]
Thanks a lot!

**Comment by selvas76** [comment]
Hi,
I am trying to configure the credential fetcher in a server without internet, and facing the proxy error while trying to build it, below is the error , could you check and let me know the options / binaries available to setup this without internet access: 


Building credentials-fetcherd binary...
go: downloading google.golang.org/grpc v1.72.0
go: downloading github.com/aws/aws-sdk-go-v2 v1.27.2
go: downloading github.com/aws/aws-sdk-go-v2/config v1.27.9
go: downloading github.com/aws/aws-sdk-go-v2/credentials v1.17.9
go: downloading github.com/coreos/go-systemd/v22 v22.5.0
go: downloading google.golang.org/protobuf v1.36.6
go: downloading github.com/aws/aws-sdk-go-v2/service/secretsmanager v1.28.5
go: downloading github.com/aws/aws-sdk-go-v2/service/s3 v1.53.0
internal/utils/aws_utils/aws_utils.go:9:2: reading github.com/aws/aws-sdk-go-v2/go.mod at revision v1.27.2: git ls-remote -q origin in /root/go/pkg/mod/cache
/vcs/0a897d51bfc9ae666ee1454be13381a5c8df2e3bb2a5bd1dba78528978aa3b56: exit status 128:
        fatal: unable to access 'https://github.com/aws/aws-sdk-go-v2/': Could not resolve host: github.com
internal/utils/aws_utils/aws_utils.go:10:2: reading github.com/aws/aws-sdk-go-v2/config/go.mod at revision config/v1.27.9: git ls-remote -q origin in /root/g
o/pkg/mod/cache/vcs/0a897d51bfc9ae666ee1454be13381a5c8df2e3bb2a5bd1dba78528978aa3b56: exit status 128:
        fatal: unable to access 'https://github.com/aws/aws-sdk-go-v2/': Could not resolve host: github.com
internal/utils/aws_utils/s3_utils.go:11:2: reading github.com/aws/aws-sdk-go-v2/service/s3/go.mod at revision service/s3/v1.53.0: git ls-remote -q origin in 
/root/go/pkg/mod/cache/vcs/0a897d51bfc9ae666ee1454be13381a5c8df2e3bb2a5bd1dba78528978aa3b56: exit status 128:
        fatal: unable to access 'https://github.com/aws/aws-sdk-go-v2/': Could not resolve host: github.com
internal/utils/aws_utils/aws_utils.go:11:2: reading github.com/aws/aws-sdk-go-v2/service/secretsmanager/go.mod at revision service/secretsmanager/v1.28.5: gi
t ls-remote -q origin in /root/go/pkg/mod/cache/vcs/0a897d51bfc9ae666ee1454be13381a5c8df2e3bb2a5bd1dba78528978aa3b56: exit status 128:
        fatal: unable to access 'https://github.com/aws/aws-sdk-go-v2/': Could not resolve host: github.com
internal/grpc/api/AddKerberosArnLease.go:14:2: reading github.com/aws/aws-sdk-go-v2/credentials/go.mod at revision credentials/v1.17.9: git ls-remote -q orig
in in /root/go/pkg/mod/cache/vcs/0a897d51bfc9ae666ee1454be13381a5c8df2e3bb2a5bd1dba78528978aa3b56: exit status 128:
